File: solutions/javascript/resistor-color/resistor-color.js
//@ts-check
//
// Helpful program so that you don't have to remember the values of the bands.
//

/** Return the value of a color band
 *
 * @param {string} color
 * @return {number}
 */
export const colorCode = (color) => COLORS.indexOf(color);

// Better Be Right Or Your Great Big Values Go Wrong
export const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white",
];

File: solutions/python/resistor-color/resistor_color.py
"""Helpful program so that you don't have to remember the values of the bands."""


def color_code(color):
    """Return the value of a color band"""
    return colors().index(color)


def colors():
    """Better Be Right Or Your Great Big Values Go Wrong"""
    return [
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white",
    ]
